awswrangler/opensearch/_write.py

```diff
@@ -556,20 +589,31 @@ def index_documents(
                 refresh_interval = _get_refresh_interval(client, index)
                 _disable_refresh_interval(client, index)
             _logger.debug("running bulk index of %s documents", len(bulk_chunk_documents))
-            _success, _errors = opensearchpy.helpers.bulk(
-                client=client,
-                actions=bulk_chunk_documents,
-                ignore_status=ignore_status,
-                chunk_size=chunk_size,
-                max_chunk_bytes=max_chunk_bytes,
-                max_retries=max_retries,
-                initial_backoff=initial_backoff,
-                max_backoff=max_backoff,
-                request_timeout=30,
+            bulk_kwargs = {
+                "ignore_status": ignore_status,
+                "chunk_size": chunk_size,
+                "max_chunk_bytes": max_chunk_bytes,
+                "max_retries": max_retries,
+                "initial_backoff": initial_backoff,
+                "max_backoff": max_backoff,
+                "request_timeout": 30,
                 **kwargs,
-            )
-            success += _success
-            errors += _errors
+            }
+            _logger.debug("running bulk with kwargs: %s", bulk_kwargs)
+            if use_threads:
+                # Parallel bulk does not support max_retries, initial_backoff & max_backoff
+                bulk_kwargs.pop("max_retries")
+                bulk_kwargs.pop("initial_backoff")
+                bulk_kwargs.pop("max_backoff")
```

Comment:
I feel like we should throw an error here instead. If a customer explicitely says they want `max_retries` and we don't support it, we shouldn't disregard that.

I think it would be fair to throw an error like:

> `max_retries` not supported when `use_threads` is set to `True`

Reply:
Fail fast and be explicit - I like that. Done.